rsocket/internal/WarmResumeManager.h

```diff
@@ -0,0 +1,98 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <deque>
+
+#include "rsocket/RSocketStats.h"
+#include "rsocket/ResumeManager.h"
+
+namespace folly {
+class IOBuf;
+}
+
+namespace rsocket {
+
+class RSocketStateMachine;
+class FrameTransport;
+
+class WarmResumeManager : public ResumeManager {
+ public:
+  explicit WarmResumeManager(
+      std::shared_ptr<RSocketStats> stats,
+      size_t capacity = DEFAULT_CAPACITY)
+      : stats_(std::move(stats)), capacity_(capacity) {}
+  ~WarmResumeManager();
+
+  void trackReceivedFrame(
+      size_t frameLength,
+      FrameType frameType,
+      StreamId streamId,
+      size_t consumerAllowance) override;
+
+  void trackSentFrame(
+      const folly::IOBuf& serializedFrame,
+      FrameType frameType,
+      folly::Optional<StreamId> streamIdPtr,
+      size_t consumerAllowance) override;
+
+  void resetUpToPosition(ResumePosition position) override;
+
+  bool isPositionAvailable(ResumePosition position) const override;
+
+  void sendFramesFromPosition(
+      ResumePosition position,
+      FrameTransport& transport) const override;
+
+  ResumePosition firstSentPosition() const override {
+    return firstSentPosition_;
+  }
+
+  ResumePosition lastSentPosition() const override {
+    return lastSentPosition_;
+  }
+
+  ResumePosition impliedPosition() const override {
+    return impliedPosition_;
+  }
+
+  void onStreamClosed(StreamId) override{
+      // No action to perform for WarmResumeManager
+  };
+
+  const std::unordered_map<StreamId, StreamResumeInfo>& getStreamResumeInfo()
+      override {
+    LOG(FATAL) << "Not Implemented for Warm Resumption";
```

Comment:
Should this crash?  It seems like we should have softer failures for someone trying to do cold resumption with `WarmResumeManager`.

Reply:
I think crash is better here.  Users should not be attempting cold resumption with `WarmResumeManager`.   This is mostly during development.  It is unlikely that a user will suddenly hit this code path and FATAL in production.